### PR TITLE
Fix TextToVideoZeroPipeline super class arguments

### LINKED ISSUE #6094: TextToVideoZeroPipeline unusable after diffusers 0.24 update
### Describe the bug

Hello diffusers team, 

First of all, thanks for your awesome work !

Here's my problem : after upgrading to diffusers 0.24, i can't succeed loading a model into the **TextToVideoZeroPipeline**.
 
.from_pretrained fails with **AttributeError: 'bool' object has no attribute '__module__'. Did you mean: '__mod__'?**

Downgrading to diffusers 0.23.1 and peft 0.7.0 to 0.6.2 make the following snippet work again.

Seems to looks like [5992](https://github.com/huggingface/diffusers/issues/5992), but with the TextToVideoZeroPipeline.

Am I the only one encountering this behavior ?

### Reproduction

```
import torch
from diffusers import TextToVideoZeroPipeline

model_id = "runwayml/stable-diffusion-v1-5"
pipe = TextToVideoZeroPipeline.from_pretrained(model_id, torch_dtype=torch.float32).to("cpu")

prompt = "A panda is playing guitar on times square"
result = pipe(prompt=prompt).images
result = [(r * 255).astype("uint8") for r in result]
imageio.mimsave("video.mp4", result, fps=4)
```

### Logs

```shell
(env) woolverine@testdeploy:~/test$ python3 ./test.py
Loading pipeline components...:   0%|                                                                                                                                                        | 0/7 [00:00<?, ?it/s]`text_config_dict` is provided which will be used to initialize `CLIPTextConfig`. The value `text_config["id2label"]` will be overriden.
`text_config_dict` is provided which will be used to initialize `CLIPTextConfig`. The value `text_config["bos_token_id"]` will be overriden.
`text_config_dict` is provided which will be used to initialize `CLIPTextConfig`. The value `text_config["eos_token_id"]` will be overriden.
Loading pipeline components...: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 7/7 [00:01<00:00,  5.95it/s]
Traceback (most recent call last):
  File "/home/woolverine/test/./test.py", line 5, in <module>
    pipe = TextToVideoZeroPipeline.from_pretrained(model_id, torch_dtype=torch.float32).to("cpu")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/woolverine/test/env/lib/python3.11/site-packages/diffusers/pipelines/pipeline_utils.py", line 1357, in from_pretrained
    model = pipeline_class(**init_kwargs)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/woolverine/test/env/lib/python3.11/site-packages/diffusers/pipelines/text_to_video_synthesis/pipeline_text_to_video_zero.py", line 314, in __init__
    super().__init__(
  File "/home/woolverine/test/env/lib/python3.11/site-packages/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py", line 236, in __init__
    self.register_modules(
  File "/home/woolverine/test/env/lib/python3.11/site-packages/diffusers/pipelines/pipeline_utils.py", line 566, in register_modules
    library = not_compiled_module.__module__.split(".")[0]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'bool' object has no attribute '__module__'. Did you mean: '__mod__'?
```

### System Info

Os : Debian 12 (VirtualBox virtual machine)
Python version : Python 3.11.2
Diffusers : 0.24
PyTorch : 2.1.0+cpu

### Who can help?

_No response_

### PR DESCRIPTION
**Pull request by Woolverine94**
# What does this PR do?

This PR fix the TextToVideoZeroPipeline in diffusers 0.24 by transforming  `safety_checker`, `feature_extractor` and `requires_safety_checker`, from positionals to keywords arguments and adding `image_encoder` to the arguments list. 

Mostly copy/pasted from a [similar PR](https://github.com/huggingface/diffusers/pull/5993) on LLMGroundedDiffusionPipeline.

Fixes #6094

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [ ] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@patrickvonplaten, @sayakpaul

### CODE REVIEW

File: src/diffusers/pipelines/text_to_video_synthesis/pipeline_text_to_video_zero.py [hunk 1]
```diff
@@ -309,10 +309,19 @@ def __init__(
...
-            vae, text_encoder, tokenizer, unet, scheduler, safety_checker, feature_extractor, requires_safety_checker
+            vae,
+            text_encoder,
+            tokenizer,
+            unet,
+            scheduler,
+            safety_checker=safety_checker,
+            feature_extractor=feature_extractor,
+            image_encoder=image_encoder,
+            requires_safety_checker=requires_safety_checker,
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/text_to_video_synthesis/pipeline_text_to_video_zero.py:324]:
```suggestion
            safety_checker=None,
            feature_extractor=None,
            image_encoder=None,
            requires_safety_checker=None,
```

They need to default to None if we want to have default values for the pipeline

**Review by patrickvonplaten — COMMENTED** [review]
cc @yiyixuxu can you take a look here?

**Comment by patrickvonplaten** [comment]
The originally PR that added the pipeline, was actually very incorrect. This PR fixes everything: https://github.com/huggingface/diffusers/pull/6213

**Comment by Woolverine94** [comment]
I can confirm (which is obviously unnecessary), that PR #6213 fixes issue #6094 (at least on my side).
Thanks a lot @patrickvonplaten  ! 🙏
I close this PR as it doesn't have meaning anymore.

**Pull request closed without merging**